### PR TITLE
Update checkstyle rules and suppressions

### PR DESCRIPTION
**Pull request by samvaity**
This PR:

- Adds new checkstyle suppression [DeclarationOrderCheck](https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/DeclarationOrderCheck.html), [EmptyCatchBlock](https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyCatchBlockCheck.html), [OverloadMethodsDeclarationOrderCheck](https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheck.html)

### CODE REVIEW

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -341,7 +361,8 @@ page at http://checkstyle.sourceforge.net/config.html -->
 
     <!-- CUSTOM CHECKS -->
     <!-- Verify the whenever a field is assigned just once in constructor to be final -->
-    <module name="com.azure.tools.checkstyle.checks.EnforceFinalFieldsCheck"/>
+    <!-- error in this custom check when using with the plugin   -->
```
Comment (samvaity) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle.xml:364]: Need to fix before check-in

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -287,6 +286,27 @@ page at http://checkstyle.sourceforge.net/config.html -->
     <module name="MissingJavadocType"/>
     <module name="MissingJavadocPackage"/>
 
+    <module name="EmptyCatchBlock">
+      <property name="exceptionVariableName" value="expected"/>
```
Comment (JonathanGiles) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle.xml:290]: I probably wouldn't enforce a name like this. `expected` doesn't read well when the exception is not expected!

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM on resourcemanager packages.

You can ping us if any issue on `sdk/<service>/azure-resourcemanager-<service>`.

@ChenTanyi @xccc-msft

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle.xml [hunk 3]
```diff
@@ -287,6 +284,25 @@ page at http://checkstyle.sourceforge.net/config.html -->
...
+  <!-- <module name="CommentsIndentation" />-->
+
+    <module name="SimplifyBooleanReturn"/>
+
+    <!--  The parts of a class or interface declaration should appear in the following order:
+      1. Class (static) variables. First the public class variables, then protected, then package level (no access modifier), and then private.
+      2. Instance variables. First the public class variables, then protected, then package level (no access modifier), and then private.
+      3. Constructors
+      4. Methods -->
+    <module name="DeclarationOrder"/>
```
Comment (alzimmermsft) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle.xml:299]: Is there any configuration options for this check? I'm wondering if we could start with a simplified version where the separate code types, class variables, instance variables, constructors, and methods, need to be group without the more intricate scoping orders.

File: sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorAsyncClient.java [hunk 4]
```diff
@@ -20,7 +20,7 @@
 /** Initializes a new instance of the asynchronous AnomalyDetectorClient type. */
 @ServiceClient(builder = AnomalyDetectorClientBuilder.class, isAsync = true)
 public final class AnomalyDetectorAsyncClient {
-    private AnomalyDetectorClientImpl serviceClient;
+    private final AnomalyDetectorClientImpl serviceClient;
```
[the remark this answers is not in the document]
Comment (alzimmermsft) [reply sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorAsyncClient.java:23]: @srnagar

File: sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorClientBuilder.java [hunk 5]
```diff
@@ -23,6 +23,16 @@
      */
     private String endpoint;
 
+    /*
+     * The HTTP pipeline to send requests through
+     */
+    private HttpPipeline pipeline;
```
[the remark this answers is not in the document]
Comment (alzimmermsft) [reply sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorClientBuilder.java:29]: @srnagar

File: sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorAsyncClient.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (srnagar) [reply sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorAsyncClient.java:23]: Yes, this should be fixed in autorest and shouldn't be updated manually.

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 6]
```diff
@@ -110,16 +121,33 @@
...
+  <!-- Fields defaultTrustManager and keyStore might be set more than once. -->
+  <suppress checks="com.azure.tools.checkstyle.checks.EnforceFinalFieldsCheck" files="com.azure.security.keyvault.jca.KeyVaultTrustManager"/>
+
+  <suppress checks="com.azure.tools.checkstyle.checks.EnforceFinalFieldsCheck" files="com.azure.storage.blob.specialized.BlobOutputStream"/>
 
   <!-- Don't apply custom Checkstyle rules to files under test package -->
   <suppress
       checks="com\.azure\.tools\.checkstyle\.checks\.(ExternalDependencyExposedCheck|NoImplInPublicAPI|ServiceClientCheck|ServiceClientBuilderCheck|ServiceInterfaceCheck|HttpPipelinePolicyCheck|JavaDocFormatting|JavadocThrowsChecks|EnforceFinalFieldsCheck|ThrowFromClientLoggerCheck|GoodLoggingCheck)"
       files=".*[/\\]src[/\\]test[/\\]java[/\\].*\.java"/>
+  <suppress checks="DeclarationOrder" files=".*[/\\]src[/\\]test[/\\]java[/\\].*\.java" />
```
[the remark this answers is not in the document]
Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:142]: Do we want this rule to run to for test folders too?

File: sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorClientBuilder.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (srnagar) [reply sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorClientBuilder.java:29]: Yeah, same here. This should be fixed in autorest. Please file GitHub issues in autorest java repo for these.

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 7]
```diff
@@ -136,43 +164,86 @@
...
+  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck" files="com.azure.messaging.eventhubs.EventHubProducerClient.java"/>
+  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck" files="com.azure.security.keyvault.administration.KeyVaultBackupAsyncClient.java"/>
+  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck" files="com.azure.security.keyvault.certificates.CertificateAsyncClient.java"/>
+  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck" files="com.azure.security.keyvault.keys.KeyAsyncClient.java"/>
+  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck" files="com.azure.security.keyvault.secrets.SecretAsyncClient.java"/>
+
+  <!-- Cryptography Client exception for service client instantiation as it provides client side crypto and is not entirely based on REST service -->
+  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck" files=".*CryptographyAsyncClient.java"/>
+
+  <!-- Return type expected to be PagedFlux but is SearchPagedFlux since it is an LRO -->
```
[the remark this answers is not in the document]
Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:223]: Copy paste 🤕

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 8]
```diff
@@ -251,156 +352,65 @@
...
+  <suppress checks="com.azure.tools.checkstyle.checks.ThrowFromClientLoggerCheck" files="com.azure.ai.formrecognizer.training.models.CopyAuthorization"/>
 
-  <!-- The variables are assigned by json creator annotation injection framework, cannot be made final. -->
-  <suppress checks="com.azure.tools.checkstyle.checks.EnforceFinalFieldsCheck" files="com.azure.ai.formrecognizer.training.models.CopyAuthorization.java"/>
-
-  <!-- Throws a non-runtime exception -->
-  <suppress checks="com.azure.tools.checkstyle.checks.ThrowFromClientLogger"
-            files="com.azure.core.http.netty.implementation.HttpProxyHandler.java"/>
+  <!-- Not neweing ClientLogger  -->
+  <suppress checks="com.azure.tools.checkstyle.checks.GoodLoggingCheck" files="com.azure.core.util.logging.ClientLogger.java"/>
```
[the remark this answers is not in the document]
Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:370]: We already are allowing for Logger instances to be static. 
I think this one is for the external logger library. I will update the comment.
image

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 9]
```diff
@@ -251,156 +352,65 @@
...
-  <!-- Suppress public/private constructor check since BlobClients need protected constructors to create EncryptedClients -->
-  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck"
-            files="com.azure.storage.blob.BlobAsyncClient.java"/>
-  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck"
-            files="com.azure.storage.blob.BlobClient.java"/>
+  <!-- Checkstyle suppression's for naming issues reported in the Synapse artifacts related clients-->
+  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck" files="com.azure.analytics.synapse.accesscontrol.RoleAssignmentsAsyncClient"/>
+  <suppress checks="DeclarationOrder" files="com.azure.analytics.synapse.accesscontrol.AccessControlClientBuilder"/>
+  <suppress checks="DeclarationOrder" files="com.azure.analytics.synapse.artifacts.ArtifactsClientBuilder"/>
+  <!-- Suppress the check on synapse code-gen classes -->
```
[the remark this answers is not in the document]
Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:391]: @jianghaolu Do you know if synapse code generation should no longer have the indentation issues?

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 10]
```diff
@@ -567,13 +504,16 @@
             files="com.azure.cosmos.ClientSideRequestStatistics"/> <!-- Need OperatingSystemMXBean from sun to obtain cpu info -->
 
   <!-- perf-test files are not public API, so they don't need to fulfill all requirements. -->
-  <suppress checks="Javadoc" files=".*[/\\]azure-storage-perf[/\\].*\.java"/>
+  <suppress checks="Javadoc" files=".*[/\\]azure-storage-perf|azure-search-perf|azure-identity-perf[/\\].*\.java"/>
```
[the remark this answers is not in the document]
Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:507]: I am not sure if Javadoc should be a requirement for perf-testing folders.
@JonathanGiles Thoughts?

File: sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorAsyncClient.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (samvaity) [reply sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorAsyncClient.java:23]: I manually updated this.
@srnagar Could you clarify on this? If this was generated code or handwritten and if it should follow the checkstyle guidelines if being generated?

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml (hunk 10 above)
[the remark this answers is not in the document]
Comment (JonathanGiles) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:507]: Yeah, no need to javadoc perf tests.

File: sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java [hunk 11]
```diff
@@ -162,15 +161,16 @@ private void initializeCryptoClients() {
      * <p><strong>Code Samples</strong></p>
      * <p>Gets the configured key in the client. Subscribes to the call asynchronously and prints out the returned key
      * details when a response has been received.</p>
-     * {@codesnippet com.azure.security.keyvault.keys.cryptography.cryptographyclient.getKey}
+     * {@codesnippet com.azure.security.keyvault.keys.cryptography.async.cryptographyclient.getKeyWithResponse}
```
[the remark this answers is not in the document]
Comment (samvaity) [reply sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java:164]: This is a reordering change. The codesnippet looks correct for its associated method.

File: sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorAsyncClient.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (srnagar) [reply sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorAsyncClient.java:23]: Anomaly detector library is completely auto-generated code. Ideally, generated code should follow the checkstyle guidelines but it currently doesn't as we use Google's code formatter for generated code. So, we are suppressing checkstyle checks for generated code. All codegen files are formatted and are consistent but it's different from the format we use.

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 12]
```diff
@@ -110,16 +121,33 @@
   <suppress checks="com.azure.tools.checkstyle.checks.EnforceFinalFieldsCheck" files=".*DeviceCodeCredentialBuilder.java"/>
   <suppress checks="com.azure.tools.checkstyle.checks.EnforceFinalFieldsCheck" files=".*InteractiveBrowserCredentialBuilder.java"/>
   <suppress checks="com.azure.tools.checkstyle.checks.EnforceFinalFieldsCheck" files=".*SearchIndexClientBuilder.java"/>
-  <suppress checks="com.azure.tools.checkstyle.checks.EnforceFinalFieldsCheck"
-            files=".*SearchServiceClientBuilder.java"/>
+  <suppress checks="com.azure.tools.checkstyle.checks.EnforceFinalFieldsCheck" files=".*SearchServiceClientBuilder.java"/>
```
[the remark this answers is not in the document]
Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:124]: Removing suppressions and clean up in the other PR https://github.com/Azure/azure-sdk-for-java/pull/18025.
This PR will only add newer checks.

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 13]
```diff
@@ -136,43 +164,86 @@
...
   <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientBuilderCheck" files=".*EventHubClientBuilder.java"/>
 
   <!-- Suppress warnings for KeyEncryptionKey and LocalCrypto client as it is a specialized client and doesn't follow the guidelines completely -->
   <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientBuilderCheck" files=".*KeyEncryptionKeyClientBuilder.java"/>
   <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientBuilderCheck" files=".*LocalCryptographyClientBuilder.java"/>
 
-  <!-- Suppress warnings for ServiceBusReceiverClientBuilder and ServiceBusSenderClientBuilder. They are created from
-  the main ServiceBusClientBuilder. -->
-  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientBuilderCheck" files="ServiceBusClientBuilder.java"/>
+  <!-- Return type expected to be PagedFlux but is PollerFLux since it is an LRO -->
```
[the remark this answers is not in the document]
Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:202]: Added in the other PR https://github.com/Azure/azure-sdk-for-java/pull/18025

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 14]
```diff
@@ -136,43 +164,86 @@
...
   <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck" files="ServiceBusSessionReceiverClient.java"/>
 
+  <!-- Event Hubs uses AMQP, which does not contain an HTTP response. Returning PagedResponse and Response does not apply. -->
+  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck"
+            files="com.azure.messaging.eventhubs.EventHubClient.java"/>
+  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck"
+            files="com.azure.messaging.eventhubs.EventHubAsyncClient.java"/>
+
+  <!-- Suppress public/private constructor check since BlobClients need protected constructors to create EncryptedClients -->
+  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck"
```
[the remark this answers is not in the document]
Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:238]: https://github.com/Azure/azure-sdk-for-java/issues/18402

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 15]
```diff
@@ -203,6 +274,37 @@
...
+
+  <!-- This class overrides a method that throws, so it cannot be avoided. -->
+  <suppress checks="com.azure.tools.checkstyle.checks.ThrowFromClientLoggerCheck" files="com.azure.security.keyvault.jca.KeyVaultTrustManager"/>
+
+  <!-- Nested blocks are still readable in the switch statements in Avro. -->
+  <suppress checks="AvoidNestedBlocks" files="com.azure.storage.internal.avro.implementation.schema.AvroSchema.java"/>
+  <suppress checks="AvoidNestedBlocks" files="com.azure.storage.internal.avro.implementation.schema.AvroType.java"/>
+
+  <!-- All auto generated files need to exclude from checkstyle -->
+  <suppress checks="UnusedImports" files=".*[/\\]storage[/\\].*Generated.*.java"/>
```
[the remark this answers is not in the document]
Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:304]: https://github.com/Azure/azure-sdk-for-java/pull/18401

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 16]
```diff
@@ -203,6 +274,37 @@
   <suppress checks="com.azure.tools.checkstyle.checks.ThrowFromClientLoggerCheck"
             files="com.azure.storage.blob.batch.BlobBatchOperationResponse.java"
             lines="110"/>
+  <!-- Throws a non-runtime exception -->
```
[the remark this answers is not in the document]
Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:277]: https://github.com/Azure/azure-sdk-for-java/issues/18399

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:223]: Removed in #18025.

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle.xml (hunk 3 above)
Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle.xml:299]: Yeah I beleive we could skip/ignore access modifiers validation. But still keep the ordering in place with this
 ```xml
<module name="DeclarationOrder">
                 <property name="ignoreModifiers" value="false" />
         </module>
```

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 17]
```diff
@@ -251,156 +352,65 @@
...
+  <suppress checks="com.azure.tools.checkstyle.checks.GoodLoggingCheck" files="com.azure.core.implementation.logging.DefaultLogger.java"/>
+  <suppress checks="com.azure.tools.checkstyle.checks.GoodLoggingCheck" files="com.azure.core.util.serializer.SerializerEncoding.java"/>
+  <suppress checks="com.azure.tools.checkstyle.checks.GoodLoggingCheck" files="com.azure.identity.DeviceCodeCredentialBuilder.java"/>
 
-  <!-- Event Hubs uses AMQP, which does not contain an HTTP response. Returning PagedResponse and Response does not apply. -->
-  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck"
-            files="com.azure.messaging.eventhubs.EventHubClient.java"/>
-  <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck"
-            files="com.azure.messaging.eventhubs.EventHubAsyncClient.java"/>
+  <suppress checks="EmptyCatchBlock" files="com.azure.core.util.serializer.JacksonAdapter.java"/>
```
[the remark this answers is not in the document]
Comment (samvaity) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:373]: https://github.com/Azure/azure-sdk-for-java/issues/18391

File: sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorAsyncClient.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (samvaity) [reply sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorAsyncClient.java]: https://github.com/Azure/azure-sdk-for-java/issues/18279

File: sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorClientBuilder.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (samvaity) [reply sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorClientBuilder.java]: https://github.com/Azure/azure-sdk-for-java/issues/18279

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 18]
```diff
@@ -469,73 +461,15 @@
...
-
-  <!-- Suppress the check on code-gen classes -->
-  <suppress checks="LineLength" files="com.azure.ai.formrecognizer.implementation.FormRecognizerClientImplBuilder"/>
-  <suppress checks="LineLength" files="com.azure.ai.formrecognizer.implementation.FormRecognizerClientImpl"/>
-  <suppress checks="LineLength" files="com.azure.ai.formrecognizer.implementation.models"/>
-  <suppress checks="LineLength" files="com.azure.ai.formrecognizer.Transforms.java"/>
-
-  <suppress checks="Indentation" files="com.azure.ai.formrecognizer.implementation.FormRecognizerClientImpl"/>
-  <suppress checks="Indentation" files="com.azure.security.keyvault.administration.implementation.RoleAssignmentsImpl"/>
-  <suppress checks="Indentation" files="com.azure.security.keyvault.administration.implementation.RoleDefinitionsImpl"/>
```
Comment (vcolin7) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:489]: Do these checks no longer fail?

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 19]
```diff
@@ -251,156 +350,65 @@
...
+  <suppress checks="EmptyCatchBlock" files="com.azure.resourcemanager.resources.fluentcore.utils.ResourceManagerUtils.java"/>
+  <!-- Too many files with ignored exception -->
+  <suppress checks="EmptyCatchBlock" files="com.azure.resourcemanager.sql.implementation.*"/>
+  <suppress checks="EmptyCatchBlock" files="com.azure.messaging.servicebus.implementation.ServiceBusReactorReceiver.java"/>
 
   <!-- TODO: Synapse access control API does not support PagedResponse and Response. -->
   <suppress checks="com.azure.tools.checkstyle.checks.ServiceClientCheck"
             files="com.azure.analytics.synapse.accesscontrol.AccessControlAsyncClient"/>
-
-  <!-- Public API already exposed: Common vocabulary errors, refer to Java Spec: https://azure.github.io/azure-sdk/java_introduction.html,
```
Comment (vcolin7) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml:337]: Was the checkstyle issue flagging these KeyVault vocabulary violations fixed?

**Review by vcolin7 — COMMENTED** [review]
Added a few comments on the checkstyle suppressions and code changes to Key Vault :)

**Comment by samvaity** [comment]
Closing this to include changes focusing on smaller areas at a time.

**Pull request closed without merging**